src/components/mal/DelMal.tsx

```diff
@@ -0,0 +1,101 @@
+import * as React from 'react';
+import './DelMal.less';
+import Systemtittel from 'nav-frontend-typografi/lib/systemtittel';
+import Normaltekst from 'nav-frontend-typografi/lib/normaltekst';
+import { useEffect, useState } from 'react';
+import { hentMal, oppdaterMal } from '../../api/api';
+import Textarea from 'nav-frontend-skjema/lib/textarea';
+import { MouseEventHandler } from 'react';
+
+export interface MalType {
+    mal: string;
+    endretAv: string;
+    dato: string;
+}
+
+function DelMal () {
+
+    const [malState, setMalState] = useState('');
+    const [feilState, setFeilState] = useState(false);
+    const [skalEndreState, setSkalEndreState] = useState(false);
+
+    useEffect(() => {
+        hentMal()
+            .then((res: MalType) => {
+                setMalState(res.mal);
+            })
+            .catch(() => {
+                setFeilState(true);
+            });
+    }, []);
+
+    if (feilState) {
+        return <div className="mal-container">Oops det skjedde en feil ...</div>;
+    }
+    
+    return(
+        <div className="del-mal">
+            {
+                !skalEndreState
+                ?
+                    (
+                        <>
+                            <Systemtittel className="del-mal-tittel">Forklaring og delmål</Systemtittel>
+                            <Normaltekst className="del-mal-beskrivelse">{malState}</Normaltekst>
+                            <DelMalKnapp
+                                tekst="Endre"
+                                onClick={(e: React.MouseEvent<HTMLElement>) => {
+                                    e.preventDefault();
+                                    setSkalEndreState(true);
+                                }}
+                            />
+                        </>
+                    )
```

Comment:
Du bør refactorere her og trekke ut de to blokkene i egne funksjoner. Logikken med "!skalEndreState" kan også trekkes ut i en egen funksjon.

Reply:
Fikset